src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -276,19 +271,18 @@ std::pair<float, float> eicrecon::TrackSeeding::findPCA(std::tuple<float,float,f
   return std::make_pair(xmin,ymin);
 }
 
-int eicrecon::TrackSeeding::determineCharge(std::vector<std::pair<float,float>>& positions) const
+int eicrecon::TrackSeeding::determineCharge(std::vector<std::pair<float,float>>& positions, float X0, float Y0, float xpos, float ypos) const
 {
   // determine the charge by the bend angle of the first two hits
   int charge = 1;
   const auto& firstpos = positions.at(0);
-  const auto& secondpos = positions.at(1);
-
-  const auto firstphi = atan2(firstpos.second, firstpos.first);
-  const auto secondphi = atan2(secondpos.second, secondpos.first);
-  auto dphi = secondphi - firstphi;
-  if(dphi > M_PI) dphi -= 2.*M_PI;
-  if(dphi < -M_PI) dphi += 2*M_PI;
-  if(dphi < 0) charge = -1;
+
+  TVector2 radial_vector( (X0-xpos), (Y0-ypos) );
+  TVector2 first_hit_vector(firstpos.first-xpos,firstpos.second-ypos);
+
+  auto del_phi = first_hit_vector.DeltaPhi(radial_vector);
```

Comment:
We don't want angle business here. Let's switch to a cross product https://root.cern.ch/doc/master/TVector2_8h_source.html#l00163
```suggestion
  double vec_prod = first_hit_vector ^ radial_vector;
```
(adjust sign accordingly)

Reply:
If we can use cross product then eigen has that functionality, in which case I propose we just switch to using `Acts::Vector2` which is already included in a different header and using these to do the matrix math which (`Acts::VectorX` is just an alias for eigen matrices)